examples/api-samples/src/electron-browser/updater/sample-updater-frontend-contribution.ts

```diff
@@ -113,14 +100,11 @@ export class SampleUpdaterFrontendContribution implements CommandContribution, M
     @inject(SampleUpdater)
     protected readonly updater: SampleUpdater;
 
-    @inject(SampleUpdaterClientImpl)
-    protected readonly updaterClient: SampleUpdaterClientImpl;
-
     protected readyToUpdate = false;
 
     @postConstruct()
     protected init(): void {
-        this.updaterClient.onReadyToInstall(async () => {
+        this.updater.onReadyToInstall(async () => {
```

Comment:
Here's an example of directly using `Event` fields directly on the proxy. Less boilerplate code than with clients.

Reply:
I generally don't mind boilerplate code if it's simpler to understand than the alternative. Typing is not the problem in software construction.

Reply:
Typing prevents mistakes. This apart, the previous required boilerplate code for the `xxxClient` API was often misused which I assumed was a symptom of a design problem.

The new API allows server to expose their events through proxies, so it almost remove the needs for clients like before. The last feature of client were requests going the other way from the main RPC setup. To do that now you can just create a new dedicated proxy.

Just as an example, the most egregious implementation I've seen using clients must be this one:

https://github.com/eclipse-theia/theia/blob/fd14c0bb423ece9add7825919cc3454875897625/packages/terminal/src/browser/terminal-frontend-contribution.ts#L199-L206

Here `TerminalWatcher` is a client to `TerminalServer`. And instead of having the remote server call a method to fetch collections from its clients and use the result, it does this roundabout procedure where it sends an event and expects the receiving side to initiate subsequent RPC calls (plural) with `this.shellTerminalServer.setCollection(...)`.

In my latest changes I replaced the idea of the client on the server side by a proxy from the frontend implementing storage methods. The server just does `environmentStore.getEnvironmentVariables()` or `environmentStore.saveEnvironmentVariables(vars)`, it is transparently handled by the frontend.